### PR TITLE
Collapse sections in options page

### PR DESCRIPTION
**Pull request by fregante**
- Includes https://github.com/refined-github/refined-github/issues/5222#issuecomment-1172237540
- In preparation for a future https://github.com/refined-github/refined-github/issues/6383

## Test URLs

Open the options page

<details>
<summary>Initial screenshots / demos </summary>

## Video

https://user-images.githubusercontent.com/1402241/220904136-ca1be78d-4b82-46b1-8faf-4277f5bfa4d7.mov

## First load

Screenshot 8

## Everything expanded
Screenshot 9

## Firefox

Screenshot 10

</details>

### CODE REVIEW

File: source/options.css [hunk 1]
```diff
@@ -34,6 +34,29 @@ li[data-validation] {
...
+	border-left: solid var(--border-left) #aaa4;
+	padding-bottom: 1px;
+}
+
+summary {
+	--summary-padding: 10px;
+	background: #aaa4;
+	list-style: none;
+	padding: var(--summary-padding);
+	cursor: pointer;
```
Comment (fregante) [inline source/options.css:48]: I'm not a super fan of the style, but I found the default `details` style inadequate for an "app UI"

Screenshot 3

**Review by yakov116 — APPROVED** [review]
Did not test it but I love it!!

File: source/options.html [hunk 2]
```diff
@@ -4,65 +4,52 @@
...
-	<hr>
+	<details id="css">
+		<summary><strong>💅 Custom CSS</strong></summary>
+		<p>Like a userstyle, useful to undo unwanted style changes</p>
+		<p><textarea name="customCSS" rows="2" spellcheck="false"></textarea></p>
+	</details>
 
-	<div>
-		<p id="show-debugging">
-			<button>Show debugging tools</button>
```
Comment (FloEdelmann) [inline source/options.html:54]: The script and styles related to this `show-debugging` button can now also be removed.

File: source/options.html [hunk 3]
```diff
@@ -86,17 +73,15 @@
...
-	<hr>
-
-	<p>
-		<label>
-			<strong class="features-header">🔋 Features</strong>
-			<input id="filter-features" type="text" placeholder="Find features" spellcheck="false" autocomplete="off" autocapitalize="off">
-		</label>
-	</p>
-	<div class="js-features"></div>
+	<details id="features">
```
Comment (FloEdelmann) [inline source/options.html:78]: Should the features section be open by default, given that it is the last section and likely the one you want to access anyway?

```suggestion
	<details id="features" open>
```

Comment (yakov116) [reply source/options.html:78]: Mixed feelings about his, I like that entire page is compact. Keeping it open takes that away.

File: source/options.css [hunk 4]
```diff
@@ -15,6 +15,15 @@
 	--rgh-red: #cf222e;
 }
 
+html:after {
+	/* Add some extra scroll to the page to reduce section toggle jumps/scroll */
+	content: '';
+	display: block;
+	height: 50vh;
+	background: url(icon.png) center no-repeat;
+	filter: opacity(0.1) saturate(0) brightness(1.2);
```
Comment (fregante) [inline source/options.css:24]: https://user-images.githubusercontent.com/1402241/221105225-7f4cc6b4-3b5d-442a-b6aa-b66c578afc46.mov

File: source/options.html [hunk 5]
```diff
@@ -4,99 +4,84 @@
...
+		<summary><strong>🔗 Refined GitHub link</strong></summary>
+		<p>This page will open when clicking the toolbar icon</p>
+		<p><input type="url" name="actionUrl" placeholder="https://github.com" spellcheck="false" autocomplete="off" autocapitalize="off"></p>
+	</details>
 
-	<div>
-		<p id="show-debugging">
-			<button>Show debugging tools</button>
+	<details id="bisect">
+		<summary><strong>🔎 Identify feature</strong></summary>
```
Comment (fregante) [inline source/options.html:53]: Extracted to its own section

Screenshot 3

File: source/options.tsx [hunk 6]
```diff
@@ -70,10 +70,15 @@ async function getTokenScopes(personalToken: string): Promise<string[]> {
...
 
+function expandTokenSection(): void {
+	select('details#token')!.open = true;
+}
+
 async function validateToken(): Promise<void> {
 	reportStatus({});
 	const tokenField = select('input[name="personalToken"]')!;
 	if (!tokenField.validity.valid || tokenField.value.length === 0) {
+		expandTokenSection();
```
Comment (fregante) [inline source/options.tsx:81]: ## Without token or invalid token: expanded

Screenshot 2

## With valid token: collapsed

Screenshot 1

## Demo of a page reload after adding a token

https://user-images.githubusercontent.com/1402241/221105827-cc948266-c68b-489a-887b-ab61200f6d89.mov

## Demo of section expansion after switching to a GHE host without token

https://user-images.githubusercontent.com/1402241/221106130-2cb4c036-26ad-4dce-8cb3-941c14a46887.mov

**Comment by fregante** [comment]
Good enough for now. If you have further suggestions, let me know

**Pull request merged**